core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -257,21 +262,56 @@ private[spark] object SizeEstimator extends Logging {
     val parent = getClassInfo(cls.getSuperclass)
     var shellSize = parent.shellSize
     var pointerFields = parent.pointerFields
+    
+    // we have only four sizes: 1, 2, 4, 8. pointerSize can be 4 or 8.
```

Comment:
I don't think you need this comment. Its pretty self explanatory that we are counting how many times each fieldSize + pointers occur